src/datanode/src/instance/sql.rs

```diff
@@ -198,6 +199,33 @@ impl Instance {
                     .execute(SqlRequest::CopyTable(req), query_ctx)
                     .await
             }
+            QueryStatement::Sql(Statement::Tql(tql)) => self.execute_tql(tql, query_ctx).await,
+        }
+    }
+
+    pub(crate) async fn execute_tql(&self, tql: Tql, query_ctx: QueryContextRef) -> Result<Output> {
+        match tql {
+            Tql::Eval(eval) => {
+                let promql = PromQuery {
+                    start: eval.start,
+                    end: eval.end,
+                    step: eval.step,
+                    query: eval.query,
+                };
+                let stmt = QueryLanguageParser::parse_promql(&promql).context(ExecuteSqlSnafu)?;
+                let logical_plan = self
+                    .query_engine
+                    .statement_to_plan(stmt, query_ctx)
+                    .context(ExecuteSqlSnafu)?;
+
+                self.query_engine
+                    .execute(&logical_plan)
+                    .await
+                    .context(ExecuteSqlSnafu)
```

Comment:
`ExecuteSqlSnafu` => `ExecuteTqlSnafu`?

Reply:
The TQL is still a SQL language, just an extension.